ruby/sdk-lightweight.md

```diff
@@ -0,0 +1,437 @@
+```
+
+#### Alternative approaches
+
+- Starting a foreign workflow from a string name
+
+```ruby
+client.start_workflow('sum-workflow', 1, 2, id: 'id-1', task_queue: 'demo')
+```
+
+- Starting a foreign workflow with the help of a stub
+
+```ruby
+class MyWorkflow < Temporal::Workflow
+  # Optionally override the name of the workflow
+  workflow 'my.workflow'
+end
+
+client.start_workflow(MyWorkflow, 1, 2, id: 'id-1', task_queue: 'demo')
+```
+
+
+## Start a worker
+
+```ruby
+require 'temporal/worker'
+require 'sum_workflow'
+require 'some_other_workflow'
+require 'sum_activity'
+require 'some_other_activity'
+
+connection = Temporal::Connection.new('localhost:7233')
+worker = Temporal::Worker.new(
+  connection,
+  namespace: 'demo',
+  task_queue: 'demo',
+  workflows: [SumWorkflow, SomeOtherWorkflow],
+  activities: [SumActivity, SomeOtherActivity]
+)
+
+worker.run
+```
+
+#### Alternative approach
+
+To support passing additional context to the activity classes a worker can be initialised with an
+instance of the activity class (only available for activities).
+
+```ruby
+redis = Redis.new('localhost:6379')
+
+Temporal::Worker.new(
+  ...,
+  activities: [CalculatorActivities.new(redis)]
+)
+```
+
+*NOTE: This however will come with a heavy warning — the activity must be strictly thread-safe as
+the provided instance will be called from multiple threads (and possibly at the same exact time).*
+
+
+## Execute a child workflow
+
+`parent_workflow.rb`
+
+```ruby
+class ChildWorkflow < Temporal::Workflow
+  def execute(name)
+    "Hello, #{name}!"
+  end
+end
+
+class ParentWorkflow < Temporal::Workflow
+  def execute
+    workflow.execute_workflow(ChildWorkflow, 'parent', id: 'id-2')
+  end
+end
+```
+
+
+## Async workflow
+
+Ruby doesn't have a standard (or at least very popular) framework for async execution of the code.
+Because of this we can assume that most Ruby engineers using the SDK won't be immediately familiar
+with writing async code. Because of this we have decided to make async execution opt-in, keeping
+everything synchronous by default.
+
+We have also chosen an eager execution pattern where any `async` block is started immediately and
+performed until blocked before proceeding with the execution. This is a bit easier to get a hold of
+and eliminates the risk of forgetting to actually start an async block. However this means that
+we'll need to find a solution for unhandled promise rejections, which we'll cover separately in a
+more detailed workflow proposal.
+
+We will introduce a helper method `async` to execute a block of code asynchronously:
+
+```ruby
+require 'sum_activity'
+
+class AsyncWorkflow < Temporal::Workflow
+  def execute
+    activity_promise = async { workflow.execute_activity(SumActivity, 1, 2) }
+    workflow_promise = async { workflow.execute_workflow(ChildWorkflow, 'parent') }
+
+    # Block execution until both promises are resolved
+    workflow.wait_for(activity_promise, workflow_promise)
```

Comment:
Is this still going to be available, or do we only want `async.all` and `async.any`?

Reply:
We also discussed `all` returning a promise that resolves with an array of results.

Reply:
Sorry, this is outdated. Pushed new version